src/manifest.rs

```diff
@@ -0,0 +1,237 @@
+use std::path::{Path, PathBuf};
+
+use crate::MANIFEST_FILE_NAME;
+use toml_edit::DocumentMut;
+
+/// Writes a package dependency to the specified TOML manifest file.
+///
+/// # Arguments
+///
+/// * `project_dir` - The directory containing the manifest file.
+/// * `package_name` - The name of the package to be added as a dependency.
+/// * `path` - The path to the package.
+///
+/// # Returns
+///
+/// A `PathBuf` representing the path to the updated manifest file.
+///
+/// # Errors
+///
+/// This function will panic if the manifest file cannot be found, if the file cannot be read,
+/// or if the content is not valid TOML.
+pub fn write_package_dep(project_dir: PathBuf, package_name: &str, path: &str) -> PathBuf {
+    let manifest = try_find_manifest(&project_dir).expect("Unable to find manifest");
+    let content = std::fs::read_to_string(manifest.clone()).expect("Cannot read file");
+    let mut doc = content.parse::<DocumentMut>().expect("Invalid TOML");
+
+    let mut table = toml_edit::InlineTable::default();
+    table.get_or_insert("path", path);
+    // Assign to the dependencies table
+    doc["dependencies"][package_name] =
```

Comment:
will it duplicate entries in dependencies if adding the same package multiple times?

Reply:
Good question! No, it doesn't duplicate. The used crate is smart enough.